pkg/azuredx/azureauth/auth.go

```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/grafana/grafana-azure-sdk-go/azsettings"
 	"github.com/grafana/grafana-azure-sdk-go/aztokenprovider"
 	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
```

Comment:
mmh, I am not sure you can do this, you will not be able to use inner packages from Grafana in plugins :thinking: Maybe you can ask in #grafana-backend or in #grafana-plugins-platform if it's possible to use backend feature flags in plugins? There may be some other way of doing it but we don't have any examples in the plugins we own.

Reply:
Yes, I had the same feeling, I was looking into external plugins on Github with no luck either.

I've removed the backend code, 
As is the branch will prevent OBO feature to be used with newly created data source when feature gate is OFF, but it will not prevent the query being done using OBO if it was enabled when the data source was created. Two ways forwards it seems:

1) _Ideal if possible_: I wonder if we can pass the `onBehalfOf = false` from the frontend when executing the query?
2) _Less user friendly_: I add code in the config editor (maybe here: https://github.com/grafana/azure-data-explorer-datasource/blob/15af456286d443f6499c714965a9ea6166420e84/src/components/ConfigEditor/ConnectionConfig.tsx#L33) to check the feature flag value. Meaning users would have to open and save the data source for the change to be applied in the data source `jsonData`